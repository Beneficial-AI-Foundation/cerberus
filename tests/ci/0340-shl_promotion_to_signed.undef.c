int main(void)
{
  unsigned int n = 1;
  unsigned char c = 1;
  n << 31; // defined (unsigned result type, so << has wrapping semantics)

  // the left operand 'c' has type 'unsigned char' which is promoted to
  // 'signed int', so the << operator has signed semantics.
  c << 30; // defined (signed result type, no overflow)
  c << 31; // UNDEFINED (signed result type, overflow!)
}
