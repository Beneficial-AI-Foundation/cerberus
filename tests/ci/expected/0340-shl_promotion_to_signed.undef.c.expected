Undefined {ub: "UB052b_non_representable_left_shift", stderr: "", loc: "<10:3--10:10>"}
